crates/interpreter/src/value.rs

```diff
@@ -0,0 +1,160 @@
+use sonatina_ir::{module::ModuleCtx, Type, I256, U256};
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Default)]
+pub enum EvalValue {
+    Literal(I256),
+    #[default]
+    Undefined,
+}
+
+impl EvalValue {
+    pub fn from_i256(i256: I256) -> Self {
+        Self::Literal(i256)
+    }
+
+    pub fn from_usize(addr: usize) -> Self {
+        Self::Literal(addr.into())
+    }
+
+    pub fn i256(&self) -> I256 {
+        let Self::Literal(i256) = *self else {
+            panic!("undefined");
+        };
+        i256
+    }
+
+    pub fn deserialize(ctx: &ModuleCtx, ty: Type, b: Vec<u8>) -> Option<Self> {
```

Comment:
```suggestion
    pub fn deserialize(ctx: &ModuleCtx, ty: Type, b: &[u8]) -> Option<Self> {
```
What about using [byteorder](https://crates.io/crates/byteorder) to avoid unnecessary heap allocation?